src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,71 @@
-import React from 'react';
+import { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { currentTodoSlice } from '../../features/currentTodo';
+
+export const TodoModal = () => {
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
```

Comment:
Consider adding a type for `state` in the `useAppSelector` hook to ensure type safety and improve code readability.